plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -0,0 +1,309 @@
+            vcs = toVcsInfo(metadata)
+        )
+    }
+
+    private fun getDependencyGraph(
+        projectDir: File,
+        depDirectives: Map<String, BazelDepDirective>,
+        lockFile: LockFile?
+    ): Set<Scope> {
+        if (lockFile != null) {
+            return parseLockFileGraph(lockFile.moduleDepGraph, depDirectives, managerName)
+        } else {
+            val modGraphProcess = run("mod", "graph", "--output", "json", workingDir = projectDir)
+            return parseBzlmodGraph(modGraphProcess.stdout, depDirectives, managerName)
+        }
+    }
+}
+
+internal fun parseLockFileGraph(
+    depGraph: Map<String, DepGraphEntry>,
+    depDirectives: Map<String, BazelDepDirective>,
+    idRefType: String
+): Set<Scope> {
+    val root = depGraph["<root>"] ?: return emptySet()
+
+    val directDeps = root.deps
+        .map { it.value }
+        // Skip Bazel internal modules that can be identified as such. Some are denoted as "bazel_tools@_".
+        .filter { !it.endsWith("@_") }
+        .toSet()
+
+    val devDeps = directDeps.filter { depDirectives[it]?.devDependency ?: false }.toSet()
+    val mainDeps = directDeps - devDeps
+
+    return setOf(
+        Scope(
+            name = "main",
+            dependencies = mainDeps.mapNotNull {
+                when (val node = depGraph[it]) {
+                    null -> null
+                    else -> parseLockFileGraphNode(node, depGraph, idRefType)
+                }
+            }.toSet()
+        ),
+        Scope(
+            name = "dev",
+            dependencies = devDeps.mapNotNull {
+                when (val node = depGraph[it]) {
+                    null -> null
+                    else -> parseLockFileGraphNode(node, depGraph, idRefType)
+                }
+            }.toSet()
+        )
+    )
+}
+
+internal fun parseLockFileGraphNode(
+    node: DepGraphEntry,
+    depGraph: Map<String, DepGraphEntry>,
+    refType: String
+): PackageReference {
+    val dependencies: Set<PackageReference> = node.deps.map {
+        it.value
+    }.filter {
+        // Skip Bazel internal modules that can be identified as such. Some are denoted as "bazel_tools@_".
+        !it.endsWith("@_")
+    }.mapNotNull {
+        when (val n = depGraph[it]) {
+            null -> null
+            else -> parseLockFileGraphNode(n, depGraph, refType)
+        }
+    }.toSet()
+
+    return PackageReference(
+        id = Identifier(
+            type = refType,
+            namespace = "",
+            name = node.name,
+            version = node.version
+        ),
+        // Static linking is the default for cc_binary targets. According to the documentation, it is off for cc_test,
+        // but according to experiments, it is on.
+        // See https://bazel.build/reference/be/c-cpp#cc_binary.linkstatic
+        linkage = PackageLinkage.STATIC,
+        dependencies = dependencies
+    )
+}
+
+internal fun parseBzlmodGraph(
```

Comment:
How about naming this `parseModuleGraph` to avoid short forms and not repeat the term Bazel as that is clear from the context of the Kotlin module.